pkg/dnet/kpfconn.go

```diff
@@ -66,20 +68,19 @@ func NewK8sPortForwardDialer(logCtx context.Context, kubeConfig *rest.Config, k8
 
 // Dial dials a port of something in the cluster.  The address format is
 // "[objkind/]objname[.objnamespace]:port".
-func (pf *k8sPortForwardDialer) Dial(ctx context.Context, addr string) (conn net.Conn, err error) {
+func (pf *k8sPortForwardDialer) Dial(ctx context.Context, addr string) (net.Conn, error) {
 	dlog.Debugf(pf.logCtx, "k8sPortForwardDialer.Dial(ctx, %q)", addr)
 
 	pod, podPortNumber, err := pf.resolve(ctx, addr)
 	if err != nil {
 		dlog.Errorf(pf.logCtx, "Error with k8sPortForwardDialer dial: %s", err)
 		return nil, err
 	}
-	inner, err := pf.dial(pod, podPortNumber)
+	conn, err := pf.dial(pod, podPortNumber)
 	if err != nil {
 		dlog.Errorf(pf.logCtx, "Error with k8sPortForwardDialer dial: %s", err)
-		return nil, err
 	}
-	return wrapUnbufferedConn(inner), nil
```

Comment:
There's a comment on `dnet.bufferedConn` explaining why a buffered connection was used here; is that no longer applicable?

Reply:
No it's not. It might be applicable in other implementations of `unbufferedConn`, but the kpfConn no longer implements that interface. It contains a `spdystream.Stream` that implements the `net.Conn` interface, so there's no need for any `Recv` or `Send`. Instead, we use that `net.Conn` directly. The gvisor network stack performs the necessary TCP buffering, so adding more buffers here is pointless.

Reply:
> The gvisor network stack performs the necessary TCP buffering, so adding more buffers here is pointless.

Makes sense, thanks.